bottomsheet/src/main/java/com/flipboard/bottomsheet/BottomSheetLayout.java

```diff
@@ -178,6 +181,35 @@ protected void onLayout(boolean changed, int left, int top, int right, int botto
         this.contentClipRect.set(0, 0, getWidth(), bottomClip);
     }
 
+    @Override
+    public boolean onKeyPreIme(int keyCode, KeyEvent event) {
+        if (keyCode == KeyEvent.KEYCODE_BACK && isSheetShowing()) {
+            if (event.getAction() == KeyEvent.ACTION_DOWN && event.getRepeatCount() == 0) {
+                KeyEvent.DispatcherState state = getKeyDispatcherState();
+                if (state != null) {
+                    state.startTracking(event, this);
+                }
+                return true;
+            } else if (event.getAction() == KeyEvent.ACTION_UP) {
+                KeyEvent.DispatcherState state = getKeyDispatcherState();
+                if (state != null) {
+                    state.handleUpEvent(event);
+                }
+                if (event.isTracking() && !event.isCanceled()) {
+                    if (isSheetShowing()) {
+                        if (getState() == BottomSheetLayout.State.EXPANDED) {
+                            peekSheet();
```

Comment:
I feel that fully dismissing should be the default, I'm not sure I can think of any use cases where you'd want that not to happen, as you start with it off screen. I guess it would make sense to go back to peeking if it was somehow locked to peek on the screen, but that isn't a feature this lib implements. I'd go with passing the existing state enum instead of a boolean `setExpanededDismissMode(State.PEEKED)` or `setExpanededDismissMode(State.HIDDEN)`.